Javadoc issues in generated code

As can be seen at https://download.eclipse.org/eclipse/downloads/drops4/I20230208-1800/compilelogs/platform.doc.isv.javadoc.txt :
```
../../../eclipse.platform.ui/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/MPart.java:182: error: unexpected end tag: </p>
	 * </p>
	   ^
../../../eclipse.platform.ui/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/MPartDescriptor.java:340: error: unexpected end tag: </p>
	 * </p>
	   ^
../../../eclipse.platform.ui/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/MPerspective.java:82: error: unexpected end tag: </p>
	 * </p>
	   ^
3 errors
```

--- a/bundles/org.eclipse.e4.ui.model.workbench/model/UIElements.ecore
+++ b/bundles/org.eclipse.e4.ui.model.workbench/model/UIElements.ecore
@@ -944,7 +944,7 @@
         <eStructuralFeatures xsi:type="ecore:EReference" name="trimBars" upperBound="-1"
             eType="#//ui/basic/TrimBar" containment="true">
           <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-            <details key="documentation" value="&lt;p>&#xD;&#xA;The collection of TrimBars associated with this part.&#xD;&#xA;@since 1.3&#xD;&#xA;&lt;/p>"/>
+            <details key="documentation" value="&lt;p>&#xA;The collection of TrimBars associated with this part.&#xA;&lt;/p>&#xA;@since 1.3"/>
           </eAnnotations>
         </eStructuralFeatures>
       </eClassifiers>
@@ -1128,7 +1128,7 @@
         <eStructuralFeatures xsi:type="ecore:EReference" name="trimBars" upperBound="-1"
             eType="#//ui/basic/TrimBar" containment="true">
           <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-            <details key="documentation" value="&lt;p>&#xD;&#xA;The collection of TrimBars associated with this perspective.&#xD;&#xA;@since 1.3&#xD;&#xA;&lt;/p>"/>
+            <details key="documentation" value="&lt;p>&#xA;The collection of TrimBars associated with this perspective.&#xA;&lt;/p>&#xA;@since 1.3"/>
           </eAnnotations>
         </eStructuralFeatures>
         <eGenericSuperTypes eClassifier="#//ui/ElementContainer">
@@ -1243,7 +1243,7 @@
         <eStructuralFeatures xsi:type="ecore:EReference" name="trimBars" upperBound="-1"
             eType="#//ui/basic/TrimBar" containment="true">
           <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-            <details key="documentation" value="&lt;p>&#xD;&#xA;The collection of TrimBars associated with the part represented by this descriptor.&#xD;&#xA;@since 1.3&#xD;&#xA;&lt;/p>"/>
+            <details key="documentation" value="&lt;p>&#xA;The collection of TrimBars associated with the part represented by this descriptor.&#xA;&lt;/p>&#xA;@since 1.3"/>
           </eAnnotations>
         </eStructuralFeatures>
       </eClassifiers>

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/MPartDescriptor.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/MPartDescriptor.java
@@ -336,8 +336,8 @@ public interface MPartDescriptor extends MApplicationElement, MUILabel, MHandler
 	 * <!-- begin-model-doc -->
 	 * <p>
 	 * The collection of TrimBars associated with the part represented by this descriptor.
-	 * @since 1.3
 	 * </p>
+	 * @since 1.3
 	 * <!-- end-model-doc -->
 	 * @return the value of the '<em>Trim Bars</em>' containment reference list.
 	 * @model containment="true"

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/impl/BasicPackageImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/impl/BasicPackageImpl.java
@@ -335,7 +335,6 @@ public class BasicPackageImpl extends EPackageImpl {
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -711,7 +710,6 @@ public class BasicPackageImpl extends EPackageImpl {
 	 * @see org.eclipse.e4.ui.model.application.descriptor.basic.MPartDescriptor#getTrimBars()
 	 * @see #getPartDescriptor()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	public EReference getPartDescriptor_TrimBars() {
@@ -1037,7 +1035,6 @@ public class BasicPackageImpl extends EPackageImpl {
 		 * <!-- begin-user-doc -->
 		 * <!-- end-user-doc -->
 		 * @since 1.3
-		 * </p>
 		 * @generated
 		 */
 		public static final EReference PART_DESCRIPTOR__TRIM_BARS = eINSTANCE.getPartDescriptor_TrimBars();

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/impl/PartDescriptorImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/descriptor/basic/impl/PartDescriptorImpl.java
@@ -369,7 +369,6 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	 * <!-- end-user-doc -->
 	 * @see #getTrimBars()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -500,7 +499,7 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	@Override
 	public List<MHandler> getHandlers() {
 		if (handlers == null) {
-			handlers = new EObjectContainmentEList<MHandler>(MHandler.class, this,
+			handlers = new EObjectContainmentEList<>(MHandler.class, this,
 					BasicPackageImpl.PART_DESCRIPTOR__HANDLERS);
 		}
 		return handlers;
@@ -514,7 +513,7 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	@Override
 	public List<MBindingContext> getBindingContexts() {
 		if (bindingContexts == null) {
-			bindingContexts = new EObjectResolvingEList<MBindingContext>(MBindingContext.class, this,
+			bindingContexts = new EObjectResolvingEList<>(MBindingContext.class, this,
 					BasicPackageImpl.PART_DESCRIPTOR__BINDING_CONTEXTS);
 		}
 		return bindingContexts;
@@ -576,7 +575,7 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	@Override
 	public List<MMenu> getMenus() {
 		if (menus == null) {
-			menus = new EObjectContainmentEList<MMenu>(MMenu.class, this, BasicPackageImpl.PART_DESCRIPTOR__MENUS);
+			menus = new EObjectContainmentEList<>(MMenu.class, this, BasicPackageImpl.PART_DESCRIPTOR__MENUS);
 		}
 		return menus;
 	}
@@ -753,7 +752,7 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	@Override
 	public List<String> getVariables() {
 		if (variables == null) {
-			variables = new EDataTypeUniqueEList<String>(String.class, this,
+			variables = new EDataTypeUniqueEList<>(String.class, this,
 					BasicPackageImpl.PART_DESCRIPTOR__VARIABLES);
 		}
 		return variables;
@@ -767,7 +766,7 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	@Override
 	public Map<String, String> getProperties() {
 		if (properties == null) {
-			properties = new EcoreEMap<String, String>(ApplicationPackageImpl.Literals.STRING_TO_STRING_MAP,
+			properties = new EcoreEMap<>(ApplicationPackageImpl.Literals.STRING_TO_STRING_MAP,
 					StringToStringMapImpl.class, this, BasicPackageImpl.PART_DESCRIPTOR__PROPERTIES);
 		}
 		return properties.map();
@@ -777,13 +776,12 @@ public class PartDescriptorImpl extends ApplicationElementImpl implements MPartD
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	@Override
 	public List<MTrimBar> getTrimBars() {
 		if (trimBars == null) {
-			trimBars = new EObjectContainmentEList<MTrimBar>(MTrimBar.class, this,
+			trimBars = new EObjectContainmentEList<>(MTrimBar.class, this,
 					BasicPackageImpl.PART_DESCRIPTOR__TRIM_BARS);
 		}
 		return trimBars;

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/MPerspective.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/MPerspective.java
@@ -78,8 +78,8 @@ public interface MPerspective
 	 * <!-- begin-model-doc -->
 	 * <p>
 	 * The collection of TrimBars associated with this perspective.
-	 * @since 1.3
 	 * </p>
+	 * @since 1.3
 	 * <!-- end-model-doc -->
 	 * @return the value of the '<em>Trim Bars</em>' containment reference list.
 	 * @model containment="true"

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/impl/AdvancedPackageImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/impl/AdvancedPackageImpl.java
@@ -610,7 +610,6 @@ public class AdvancedPackageImpl extends EPackageImpl {
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -1358,7 +1357,6 @@ public class AdvancedPackageImpl extends EPackageImpl {
 	 * @see org.eclipse.e4.ui.model.application.ui.advanced.MPerspective#getTrimBars()
 	 * @see #getPerspective()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	public EReference getPerspective_TrimBars() {
@@ -1621,7 +1619,6 @@ public class AdvancedPackageImpl extends EPackageImpl {
 		 * <!-- begin-user-doc -->
 		 * <!-- end-user-doc -->
 		 * @since 1.3
-		 * </p>
 		 * @generated
 		 */
 		public static final EReference PERSPECTIVE__TRIM_BARS = eINSTANCE.getPerspective_TrimBars();

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/impl/PerspectiveImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/advanced/impl/PerspectiveImpl.java
@@ -239,7 +239,6 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	 * <!-- end-user-doc -->
 	 * @see #getTrimBars()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -273,7 +272,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public List<MPartSashContainerElement> getChildren() {
 		if (children == null) {
-			children = new EObjectContainmentWithInverseEList<MPartSashContainerElement>(
+			children = new EObjectContainmentWithInverseEList<>(
 					MPartSashContainerElement.class, this, AdvancedPackageImpl.PERSPECTIVE__CHILDREN,
 					UiPackageImpl.UI_ELEMENT__PARENT) {
 				private static final long serialVersionUID = 1L;
@@ -428,7 +427,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public List<String> getVariables() {
 		if (variables == null) {
-			variables = new EDataTypeUniqueEList<String>(String.class, this,
+			variables = new EDataTypeUniqueEList<>(String.class, this,
 					AdvancedPackageImpl.PERSPECTIVE__VARIABLES);
 		}
 		return variables;
@@ -442,7 +441,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public Map<String, String> getProperties() {
 		if (properties == null) {
-			properties = new EcoreEMap<String, String>(ApplicationPackageImpl.Literals.STRING_TO_STRING_MAP,
+			properties = new EcoreEMap<>(ApplicationPackageImpl.Literals.STRING_TO_STRING_MAP,
 					StringToStringMapImpl.class, this, AdvancedPackageImpl.PERSPECTIVE__PROPERTIES);
 		}
 		return properties.map();
@@ -456,7 +455,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public List<MHandler> getHandlers() {
 		if (handlers == null) {
-			handlers = new EObjectContainmentEList<MHandler>(MHandler.class, this,
+			handlers = new EObjectContainmentEList<>(MHandler.class, this,
 					AdvancedPackageImpl.PERSPECTIVE__HANDLERS);
 		}
 		return handlers;
@@ -470,7 +469,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public List<MBindingContext> getBindingContexts() {
 		if (bindingContexts == null) {
-			bindingContexts = new EObjectResolvingEList<MBindingContext>(MBindingContext.class, this,
+			bindingContexts = new EObjectResolvingEList<>(MBindingContext.class, this,
 					AdvancedPackageImpl.PERSPECTIVE__BINDING_CONTEXTS);
 		}
 		return bindingContexts;
@@ -484,7 +483,7 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	@Override
 	public List<MWindow> getWindows() {
 		if (windows == null) {
-			windows = new EObjectContainmentEList<MWindow>(MWindow.class, this,
+			windows = new EObjectContainmentEList<>(MWindow.class, this,
 					AdvancedPackageImpl.PERSPECTIVE__WINDOWS);
 		}
 		return windows;
@@ -494,13 +493,12 @@ public class PerspectiveImpl extends ElementContainerImpl<MPartSashContainerElem
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	@Override
 	public List<MTrimBar> getTrimBars() {
 		if (trimBars == null) {
-			trimBars = new EObjectContainmentEList<MTrimBar>(MTrimBar.class, this,
+			trimBars = new EObjectContainmentEList<>(MTrimBar.class, this,
 					AdvancedPackageImpl.PERSPECTIVE__TRIM_BARS);
 		}
 		return trimBars;

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/MPart.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/MPart.java
@@ -178,8 +178,8 @@ public interface MPart extends MPartSashContainerElement, MStackElement, MContri
 	 * <!-- begin-model-doc -->
 	 * <p>
 	 * The collection of TrimBars associated with this part.
-	 * @since 1.3
 	 * </p>
+	 * @since 1.3
 	 * <!-- end-model-doc -->
 	 * @return the value of the '<em>Trim Bars</em>' containment reference list.
 	 * @model containment="true"

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/impl/BasicPackageImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/impl/BasicPackageImpl.java
@@ -712,7 +712,6 @@ public class BasicPackageImpl extends EPackageImpl {
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -1093,7 +1092,6 @@ public class BasicPackageImpl extends EPackageImpl {
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -3494,7 +3492,6 @@ public class BasicPackageImpl extends EPackageImpl {
 	 * @see org.eclipse.e4.ui.model.application.ui.basic.MPart#getTrimBars()
 	 * @see #getPart()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	public EReference getPart_TrimBars() {
@@ -4098,7 +4095,6 @@ public class BasicPackageImpl extends EPackageImpl {
 		 * <!-- begin-user-doc -->
 		 * <!-- end-user-doc -->
 		 * @since 1.3
-		 * </p>
 		 * @generated
 		 */
 		public static final EReference PART__TRIM_BARS = eINSTANCE.getPart_TrimBars();

--- a/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/impl/PartImpl.java
+++ b/bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/application/ui/basic/impl/PartImpl.java
@@ -371,7 +371,6 @@ public class PartImpl extends UIElementImpl implements MPart {
 	 * <!-- end-user-doc -->
 	 * @see #getTrimBars()
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 * @ordered
 	 */
@@ -768,7 +767,6 @@ public class PartImpl extends UIElementImpl implements MPart {
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
 	 * @since 1.3
-	 * </p>
 	 * @generated
 	 */
 	@Override

